Source/CronosPlayUnreal/Private/DynamicContractActor.cpp

```diff
@@ -0,0 +1,241 @@
+// Copyright 2022, Cronos Labs. All Rights Reserved
+
+#include "DynamicContractActor.h"
+
+// Sets default values
+ADynamicContractActor::ADynamicContractActor() {
+    // Set this actor to call Tick() every frame.  You can turn this off to
+    // improve performance if you don't need it.
+    PrimaryActorTick.bCanEverTick = false;
+    defiWallet = NULL;
+    _coreContract = NULL;
+}
+
+void ADynamicContractActor::Destroyed() {
+    Super::Destroyed();
+
+    if (GEngine)
+        GEngine->AddOnScreenDebugMessage(
+            -1, 15.0f, FColor::Yellow,
+            TEXT("ADynamicContractActor  Destroyed"));
+
+    DestroyContract();
+
+    assert(NULL == _coreContract);
+}
+
+void ADynamicContractActor::DestroyContract() {
+    if (_coreContract != NULL) {
+        // restored back
+        rust::cxxbridge1::Box<org::defi_wallet_core::EthContract> tmpwallet =
+            rust::cxxbridge1::Box<org::defi_wallet_core::EthContract>::from_raw(
+                _coreContract);
+        _coreContract = NULL;
+    }
+}
+// Called when the game starts or when spawned
+void ADynamicContractActor::BeginPlay() { Super::BeginPlay(); }
+
+// Called every frame
+void ADynamicContractActor::Tick(float DeltaTime) { Super::Tick(DeltaTime); }
+
+void ADynamicContractActor::NewEthContract(FString cronosrpc,
```

Comment:
nope, because only signing call needs DefiWallet

Reply:
I mean, the cronosrpc parameter is not shared with DefiWalletCoreActor, users have to pass the RPC to the function again. Any ways to share RPC, so that users don't need to bother the rpc setting again? Related issue: https://github.com/cronos-labs/play-unreal-plugin/issues/187

Reply:
for call, encode,  DynamicContract don't need DefiWalletActor.